### PR TITLE
Fix unsigned types and "infinite" array definitions

### PR DESCRIPTION
**Pull request by johnnovak**

### CODE REVIEW

**Comment by Araq** [comment]
This is a breaking change with unclear advantages.

**Comment by johnnovak** [comment]
The advantage is that now there's a direct mapping between unsigned types defined in the Windows headers and the Nim types, so there's no need to cast.

Also, the `array[0..0, type]` form doesn't work anymore with the current version of Nim (without trickery). I can't index those arrays because the compiler (correctly) thinks they are of zero length. I guess that was never the intention? I'm guessing bounds checking should be somehow disabled for those types and this is one way of doing it.

**Comment by Araq** [comment]
The unsigned changes are fine but I'm worried that the arbitrary array sizes mess up with ``sizeof`` and friends. Nim supports an ``.unchecked`` pragma for arrays.

**Comment by johnnovak** [comment]
Please have a look now, I've converted it to use the `unchecked` pragma. By the way, I learned that large-array-hack from the standard library, I think it was your code, but I agree it's much cleaner this way.

File: oldwinapi.nimble [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [Package]
 name          = "oldwinapi"
-version       = "1.0.0"
+version       = "1.0.1"
```
Comment (Araq) [inline oldwinapi.nimble]: Nice that you update the version, but it's a breaking change and so should be "1.1.0". :-)

File: windows.nim [hunk 2]
```diff
@@ -12,6 +12,9 @@
 
 {.deadCodeElim: on.}
 {.push gcsafe.}
+
+type UncheckedArray {.unchecked.} [T] = array[0..0, T]
```
Comment (Araq) [inline windows.nim]: We have this in ``system.nim`` nowadays. But for older nims do this:

```nim
when not declared(UncheckedArray):
  type UncheckedArray {.unchecked.} [T] = array[0..0, T]
```

File: oldwinapi.nimble (hunk 1 above)
Comment (dom96) [reply oldwinapi.nimble]: Actually it should be "2.0.0".

**Comment by dom96** [comment]
Also, since we're doing breaking changes in this PR anyway we should make it Nimble-compatible (by moving all modules into an `oldwinapi` directory).

**Comment by johnnovak** [comment]
All good suggestions guys, I've done the changes. While we're in breaking changes land, is the "old" prefix really needed in the project name? Why not just rename it to "winapi" while we're at it?

**Comment by dom96** [comment]
Sure, I'm happy with that.

**Comment by johnnovak** [comment]
Okay, are you guys gonna rename the repo to "winapi" and change the project description? I'll do the change in the PR and create a new PR in the Nimble packages project.

**Comment by Araq** [comment]
Sorry, but you cannot rename it to ``winapi``, ``winapi`` is reserved for a new re-implementation that uses a different license (!).

**Comment by johnnovak** [comment]
No problem, I've reverted the rename commit. It's ready for merge.

File: oldwinapi/windows.nim [hunk 3]
```diff
@@ -8357,14 +8361,14 @@ type
...
+    dbcp_name*: UncheckedArray[char]
 
   PDEV_BROADCAST_PORT* = ptr DEV_BROADCAST_PORT
   TDEVBROADCASTPORT* = DEV_BROADCAST_PORT
   DEV_BROADCAST_USERDEFINED* {.final, pure.} = object
     dbud_dbh*: DEV_BROADCAST_HDR
-    dbud_szName*: array[0..0, char]
-    dbud_rgbUserDefined*: array[0..0, int8]
+    dbud_szName*: UncheckedArray[char]
+    dbud_rgbUserDefined*: UncheckedArray[int8]
```
Comment (Araq) [inline oldwinapi/windows.nim]: This field is impossible to use and according to https://msdn.microsoft.com/en-us/library/windows/desktop/aa363481(v=vs.85).aspx does not exist.

File: oldwinapi/windows.nim [hunk 4]
```diff
@@ -9136,8 +9140,8 @@ type
     rclBounds*: RECTL
     nPolys*: DWORD
     cptl*: DWORD
-    aPolyCounts*: array[0..0, DWORD]
-    aptl*: array[0..0, POINTL]
+    aPolyCounts*: UncheckedArray[DWORD]
+    aptl*: UncheckedArray[POINTL]
```
Comment (Araq) [inline oldwinapi/windows.nim:9143]: This cannot be correct either. Only the last field in an object can be an UncheckedArray.

File: oldwinapi/windows.nim [hunk 5]
```diff
@@ -9149,8 +9153,8 @@ type
     rclBounds*: RECTL
     nPolys*: DWORD
     cpts*: DWORD
-    aPolyCounts*: array[0..0, DWORD]
-    apts*: array[0..0, POINTS]
+    aPolyCounts*: UncheckedArray[DWORD]
+    apts*: UncheckedArray[POINTS]
```
Comment (Araq) [inline oldwinapi/windows.nim:9156]: This cannot be correct either. Only the last field in an object can be an UncheckedArray.

**Comment by johnnovak** [comment]
Yeah, I can remove these for sure. Btw, I just mechanically replaced all occurrences, I assumed the definitions were otherwise correct... I'll do this tomorrow.

File: oldwinapi/windows.nim (hunk 5 above)
Comment (johnnovak) [reply oldwinapi/windows.nim:9156]: `apts` and `abTypes` are just pointers to "flexible arrays", so I think this is fine.
https://msdn.microsoft.com/en-us/library/windows/desktop/dd162560(v=vs.85).aspx

File: oldwinapi/windows.nim (hunk 4 above)
Comment (johnnovak) [reply oldwinapi/windows.nim:9143]: `aptl` and `abTypes` are just pointers to "flexible arrays", so I think this is fine.
https://msdn.microsoft.com/en-us/library/windows/desktop/dd162558(v=vs.85).aspx

File: oldwinapi/windows.nim (hunk 3 above)
Comment (johnnovak) [reply oldwinapi/windows.nim]: Removed.

File: oldwinapi/windows.nim (hunk 5 above)
Comment (Araq) [reply oldwinapi/windows.nim:9156]: Very strange, but ok.

**Pull request merged**